dashboard/client/src/components/TaskTable.tsx

```diff
@@ -330,11 +337,29 @@ const TaskTableActions = ({ task }: TaskTableActionsProps) => {
       ? `Error Type: ${task.error_type}\n\n${task.error_message}`
       : undefined;
 
+  const isTaskActive = task.state === "RUNNING" && task.worker_id;
+
   return (
     <React.Fragment>
       <Link component={RouterLink} to={`tasks/${task.task_id}`}>
         Log
       </Link>
+      {isTaskActive && (
+        <React.Fragment>
+          <br />
+          <TaskCpuProfilingLink
+            taskId={task?.task_id}
```

Comment:
are the `?` in `task?` necessary? 
Same with all the other places in this file.

Reply:
Good catch! I tried to prevent accessing an empty object by default. However, after looking into the type Task, I noticed that task_id and other properties are not optional. Therefore, we can access task.task_id directly.